### PR TITLE
Adds cassandra3

### LINKED ISSUE #44: add support for cassandra3
cassandra3 is an incubating project for cassandra v3.9+

It writes a list of dependency links directly, which is nice as it no longer needs the deprecated "Dependencies" envelope type. However, parsing must be completely different as UDTs are in use.

Please remember that this project is JRE 7+ when authoring code.

### PR DESCRIPTION
**Pull request by adriancole**
Fixes #44

Note: there are some test-only workaround for UDTs, which hopefully we can at some point revert.

### CODE REVIEW

File: cassandra3/src/main/java/zipkin/dependencies/cassandra3/CassandraDependenciesJob.java [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+        "Running Dependencies job for {}: {} ≤ Span.timestamp {}",
+        dateStamp,
+        microsLower,
+        microsUpper);
+
+    SparkContext sc = new SparkContext(conf);
+    try {
+      JavaRDD<DependencyLink> links = javaFunctions(sc)
+          .cassandraTable(keyspace, "span")
+          .map(CassandraRowToSpan.INSTANCE)
```
Comment (adriancole) [inline cassandra3/src/main/java/zipkin/dependencies/cassandra3/CassandraDependenciesJob.java]: @llinder @michaelsembwever 

TL;DR; it would be more convenient if span.trace_id was always the lower 64 bits (right-most 16 chars) instead of 16 or 32 depending on length.

In our new schema, we store the 32chars as trace_id if strictTraceId=true. This creates a bit of a problem in spark as data with different trace IDs aren't "near eachother", so when scanning in mixed mode (default for spark), so we have decode and distribute more data.

For example, to use groupBy, we have to map every span into an object like you see above. I'm thinking it might be more practical to always have span.trace_id as the same as it was before traceIdLow, then partitions will work naturally and especially larger sites can process faster.. Reason is that for a long time there will be mixed bit length and unless we accommodate this one way or another, the dependency tree will be messed up.

wdyt?

https://github.com/datastax/spark-cassandra-connector/blob/master/doc/3_selection.md#grouping-rows-by-partition-key

File: cassandra3/src/main/java/zipkin/dependencies/cassandra3/CassandraDependenciesJob.java [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+    log.info(
+        "Running Dependencies job for {}: {} ≤ Span.timestamp {}",
+        dateStamp,
+        microsLower,
+        microsUpper);
+
+    SparkContext sc = new SparkContext(conf);
+    try {
+      JavaRDD<DependencyLink> links = javaFunctions(sc)
+          .cassandraTable(keyspace, "span")
```
Comment (adriancole) [inline cassandra3/src/main/java/zipkin/dependencies/cassandra3/CassandraDependenciesJob.java]: TODO: suppose we could limit this to data stored an hour before and 5m after the window (ex to decode and filter less data)

File: cassandra3/src/main/java/zipkin/dependencies/cassandra3/CassandraDependenciesJob.java (hunk 1 above)
Comment (adriancole) [reply cassandra3/src/main/java/zipkin/dependencies/cassandra3/CassandraDependenciesJob.java]: another way is we write the job in 2 different ways, then look at the `STRICT_TRACE_ID` variable. if `STRICT_TRACE_ID=false`, then we do the more expensive groupBy. I think I'll go that way honestly as people can always re-process a day's data if they found themselves in this situation

Comment (llinder) [reply cassandra3/src/main/java/zipkin/dependencies/cassandra3/CassandraDependenciesJob.java]: Not sure I full understand the tradeoffs with only storing the lower 64 bits in `span.trace_id` but it sounds reasonable.   The second option seems like it would be acceptable as well but I'm probably biased on that decision because I'm using strictTraceId=true.

File: cassandra3/src/main/java/zipkin/dependencies/cassandra3/CassandraDependenciesJob.java (hunk 2 above)
Comment (llinder) [reply cassandra3/src/main/java/zipkin/dependencies/cassandra3/CassandraDependenciesJob.java]: Seem like a really good performance improvement for folks with lots of data.  The current spark job doesn't run in a realistic amount of time on our current dataset.  Limiting to a subset of the data would be a big win.  Probably an improvement for a later time though?

**Comment by adriancole** [comment]
ok all works
